parser/src/parser.rs

```diff
@@ -81,9 +81,6 @@ impl<'slice, 'input> BlockParser<'slice, 'input> {
     pub fn parse(mut parser: Self) -> (Self, Blocks) {
         let mut blocks = Vec::default();
 
-        #[cfg(debug_assertions)]
-        let mut curr_len = parser.iter.max_len();
-
```

Comment:
This caused problems in case an ambiguous token was split,
because I cache the split token, which means the underlying iterator does not make progress for one iteration.

The block parser should always make progress, so it might be ok to keep it in the block parser loop. For now I decided to remove it to not get conflicts for fast edits in LSP mode.